src/main/java/fi/vm/yti/datamodel/api/v2/service/SearchIndexService.java

```diff
@@ -55,14 +56,56 @@ public CountSearchResponse getCounts(ModelSearchRequest searchRequest, YtiUser u
         return ModelQueryFactory.parseModelCountResponse(client.searchResponse(query, IndexModel.class));
     }
 
-    public SearchResponseDTO<IndexModel> searchModels(ModelSearchRequest request, YtiUser user) {
+    public SearchResponseDTO<ModelSearchResultDTO> searchModels(ModelSearchRequest request, YtiUser user) {
 
         if (!user.isSuperuser()) {
             request.setIncludeDraftFrom(groupManagementService.getOrganizationsForUser(user));
         }
 
+        // search resources with the same parameters
+        var resourceSearchRequest = new ResourceSearchRequest();
+        resourceSearchRequest.setQuery(request.getQuery());
+        // search for status as well?
+
+        var resourceQuery = ResourceQueryFactory.createResourceQuery(
+                resourceSearchRequest,
+                Collections.emptyList(),
+                Collections.emptyList());
+        var resources = client.search(resourceQuery, IndexResource.class);
+
+        // now let's add those matching models to the model search request
+        request.setAdditionalModelIds(resources
+                .getResponseObjects()
+                .stream()
+                // model id should include the version, if available
+                .map(r -> {
+                    var id = r.getIsDefinedBy();
+                    if (r.getFromVersion() != null) {
+                        id = id + r.getFromVersion() + ModelConstants.RESOURCE_SEPARATOR;
+                    }
+                    return id;
+                })
+                .filter(Objects::nonNull)
+                .map(DataModelURI::fromURI)
+                .map(DataModelURI::getGraphURI)
```

Comment:
This mapping is redundant, because mapped id (starting at line 82) is graph's URI